### ISSUE TITLE
`connectNulls` with stepLine fails when after a long number of `null` entries occur

### ISSUE THREAD

**Issue by atanasbozhkov** [apache/echarts#11274]
### Version
4.2.1

### Reproduction link
[https://jsfiddle.net/peyuof13/2/](https://jsfiddle.net/peyuof13/2/)

### Steps to reproduce
1. Create a line step chart.
2. Set series data containing some starting values followed by a number of 'null' entries.
3. Set `connectNulls` to `true`on that series.

### What is expected?
E-charts should draw a draw a long-step line from the start to the end of the chart.

### What is actually happening?
After a long-null range of series occurs - echarts reverts to drawing an interpolated line.

![Step Line](https://user-images.githubusercontent.com/1144752/65150362-9613d400-d9e9-11e9-8f5f-084ef05d8ab3.png)

**Comment by Ovilia** [comment]
@atanasbozhkov Thanks for reporting. I think it makes sense for step line charts to have step lines with `connectNulls`.

**Comment by atanasbozhkov** [comment]
I think I have found the cause.  The problem seems to be in the `turnPointsIntoStep` func in [LineView.js](https://github.com/apache/incubator-echarts/blob/fd064123626c97b36cbd6da1b5fc73385c280abd/src/chart/line/LineView.js#L189)

The function automatically grabs the next point without checking if any of the point coordinates are NaN - which ends up with a bad point array.

The way I see it this can be fixed in two ways - 
1. We can sanitize the points array at the beginning of the function by filtering all NaN values. 
Something like this would do it:
`points = points.filter((el) => !isNaN(el[0]) && !isNaN(el[1]));` 

2. In the actual for-loop we can skip over NaN elements until we get the next well-defined point. 
```
 for (var i = 0; i < points.length - 1; i++) {
        var pt = points[i];
        // Skip over undefined points.
        while (isNaN(points[i + 1][0]) || isNaN(points[i + 1][1])) {
            i++;
        }
        var nextPt = points[i + 1];

```
Running a dev build locally - and testing with the reproduction code I posted above seems to result in a nice step line. 

Let me know which way you think is more suitable or if there's something I'm missing. 
I'm happy to raise a PR and add tests if necessary.
